### DISCUSSION TITLE
Apparent problem with ncell

### DISCUSSION THREAD

**Discussion by laurapaccini** [tobac-project/tobac#245] [Q&A] [answered]
Hi, 
I've been trying to apply tobac to my simulations of precipitation, but I have some doubts about "ncells" output. 
I only use one threshold with target 'maximum', which are the same in the feature detection and segmentation functions. I also set 'n_min_threshold' to be relatively large (e.g 200) so it only detects features with a minimum of ncells. However, after performing the segmentation function I get many values with ncells << than my 'n_min_threshold, and even with zero values. The only option to avoid the ncell=0 is by setting position_threshold='extreme', but in that case the number of cases with ncells >200 is less than with the other position_threshold options (e.g weighted_diff). 

Here is an example of such case (ncell =0 ) with every option except with position_threshold='extreme':

![image](https://user-images.githubusercontent.com/20461865/218386622-67667750-7a8b-4780-89e0-cdef42918440.png)

The other issue is that the mask output from segmentation does not always coincide with the ncells. I saw this especially when the ncells are < 200. For example, for a given timestep the dataframe shows features with ncells = 21. However, when looking at the mask, some are absent, others have sizes much larger than those correspondent to 21.

<img width="962" alt="Screen Shot 2023-02-13 at 2 00 36 AM" src="https://user-images.githubusercontent.com/20461865/218391835-5b9e63df-d832-4cd8-b8c0-3ea7de7cf522.png">

<img width="919" alt="Screen Shot 2023-02-13 at 2 03 04 AM" src="https://user-images.githubusercontent.com/20461865/218392286-779f0a57-5bc1-43aa-b522-b51b04382348.png">

Here the plot correspond to features 30 and 31:

![image](https://user-images.githubusercontent.com/20461865/218393102-2274cdb7-4b0c-4c1c-8df2-01dc9ea34996.png)

I created a repository in my account with the scripts and subsets of my data in case you want to reproduce these results.  (https://github.com/laurapaccini/test_tobac)

**Comment by JuliaKukulies** [comment, marked as answer]
Thank you for the careful report of this, @laurapaccini!  

The first issue is a behavior of the segmentation in `tobac` that we have recognized before and described in more detail on our documentation page: https://tobac.readthedocs.io/en/latest/features_without_segmented_area.html As you found out correctly, this behavior can be avoided when setting `position_threshold='extreme'`. Alternatively, you could set the segmentation threshold slightly higher than your feature detection threshold. This is because the feature detection will detect features that are `equal to or larger` than your threshold, while the segmentation procedure only includes areas `larger than` the segmentation threshold. 

The second issue looks like a potential bug to me. I will have a closer look into your example notebook and move this question to our  [issues](https://github.com/tobac-project/tobac/issues) !

**Reply by JuliaKukulies** [reply]
OK @laurapaccini, I could reproduce your results and found that the second issue is related to a bug that assigns the value for `ncells` of the last feature in the timestep to all features in that timestep. I raised an issue for that (see #246) and we will fix this bug asap. Thanks again for reporting this!

**Reply by laurapaccini** [reply]
Changing the threshold value sounds like a good option! Thank you as well for looking into the segmentation issue!
